game_patch/object/entity.cpp

```diff
@@ -201,6 +202,79 @@ CodeInjection entity_process_pre_hide_riot_shield_injection{
     },
 };
 
+FunHook<void(int)> entity_blood_throw_gibs_hook{
+    0x0042E3C0, [](int handle) {
+        if (g_game_config.gibbing && !rf::is_multi) {
+            rf::Object* objp = rf::obj_from_handle(handle);
+
+            if (!objp) {
+                return; // invalid object
+            }
+
+            rf::Vector3 pos = objp->pos;
+            rf::GRoom* room = objp->room;
+            int explosion_vclip = 30;
+            int chunk_explosion_vclip = 30;
+            float explosion_vclip_radius = 1.0f;
+            rf::String debris_filename = "df_meatchunks0.V3D";
+
+            int debris_max_lifetime = 7000;
+            float debris_velocity = 8.5f;
+            float damage_scale = 1.0f;
+            rf::String cust_snd_set = "gib bounce";
+
+            if (objp->type == rf::OT_ENTITY) {
+                rf::Entity* ep = static_cast<rf::Entity*>(objp);
+
+                explosion_vclip = (ep->info->explode_vclip_index < 0)
+					? ep->info->explode_vclip_index : explosion_vclip;
+
+                explosion_vclip_radius = (ep->info->explode_vclip_radius > 0.0f)
+					? ep->info->explode_vclip_radius : explosion_vclip_radius;
+
+                debris_filename = (!ep->info->debris_filename.empty())
+					? ep->info->debris_filename : debris_filename;
+            }
+            else if (objp->type = rf::OT_CORPSE) {
+            }
+            else {
+                return;
+            }
+
+            rf::game_do_explosion(
+                explosion_vclip, room, 0, &pos, explosion_vclip_radius, damage_scale, 0);
+
+            rf::debris_spawn_from_object(
+                objp, debris_filename, chunk_explosion_vclip, debris_max_lifetime, debris_velocity, &cust_snd_set);
+        }
+    }
+};
+
+CodeInjection entity_damage_explosive_death_injection{
+    0x0041A413,
+    [](auto& regs) {        
+        rf::Entity* ep = regs.esi;
+        if (g_game_config.gibbing && !rf::is_multi && ep) {
+            int damage_type = regs.ebp;
+            ep->ai.explosive_last_damage = (damage_type == 3);
+            if (ep->ai.explosive_last_damage && ep->info->body_temp >= 90.0f) {
```

Comment:
this body temp check deserve a comment maybe, but still I'm not sure it's a good way to approach this. I've read something that you wanted to check "humanoid" flag but I think it's not necessary a good thing to check. But what about checking if material is flash?

Reply:
Basing it on `body_temp` appears, based on my analysis of the stock game files, the only realistic way to ensure this behaves properly in the stock game. The only entities that have `body_temp` > 90 (and would therefore gib) are humanoid characters and the reeper/baby_reeper. This avoids a situation where entities that have custom death animations and/or are enormous (like the rock snake) but have material flesh are gibbed. `body_temp` perfectly limits the entities affected in the stock game and provides a very straightforward and sensical way for custom mod authors to make custom entities in their mod compatible with this feature.